mantidproject/mantid
# A 'moving window average' mode for `CalculateFlatBackground`

This PR implements a _moving window average_ mode for the [`CalculateFlatBackground`](http://docs.mantidproject.org/nightly/algorithms/CalculateFlatBackground-v1.html) algorithm. The new mode calculates a window average for each point in the spectra and takes the minimum as the flat background value. This is handy for TOF spectra acquired at ILL when there is no well-defined background plateau.

Resolves #17584 .

**To test:**

Try the algorithm on some real data and check that it returns sane results. With _AveragingWindowWidth_ set to 1 it should give just the minimum value. Note that the algorithm uses cyclic boundary conditions when calculating the moving average. The two original modes of operation (_Linear Fit_ and _Mean_) should work exactly as they used to except that _XMin_ and _XMax_ are not validated until the algorithm is run.

_Release notes have been updated accordingly._

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
